src/Param.h

```diff
@@ -56,7 +48,8 @@ class Param {
 
     /** Construct a scalar parameter of type T an initial value of
```

Comment:
I'm not sure about this comment. Is it trying to use the word "scalar" to mean "not a pointer?" I don't see any logic here that prevents it matching on vector types if we mapped them to some C++ equivalent.